grpc/src/test/java/com/linecorp/armeria/server/grpc/GrpcServiceImplErrorTest.java

```diff
@@ -0,0 +1,303 @@
+                            GrpcService.builder()
+                                       .addService(service)
+                                       .intercept(corruptedServerInterceptor)
+                                       .build());
+            sb.decorator(LoggingService.newDecorator());
+        }
+    };
+
+    // Normal case of #onError at Unary RPC, but metadata set in server interceptor is corrupted.
+    // Client cannot expect corrupted metadata is returned from server.
+    @Test
+    void clientUnaryCall2ForServerUsingCorruptedInterceptor() throws InterruptedException {
+        try (ClientRequestContextCaptor clientCaptor = Clients.newContextCaptor()) {
+            assertThatThrownBy(() -> {
+                final TestServiceBlockingStub client =
+                        GrpcClients.builder("http://127.0.0.1:" + server.httpPort())
+                                   .pathPrefix("/corruptedInterceptor")
+                                   .build(TestServiceBlockingStub.class);
+                client.unaryCall2(REQUEST_MESSAGE);
+            }).satisfies(cause -> {
+                assertThat(Status.fromThrowable(cause).getCode()).isEqualTo(Status.INTERNAL.getCode());
+                assertThat(Status.trailersFromThrowable(cause)).satisfies(metadata -> {
+                    assertThat(metadata.keys()).doesNotContain(KEY_OF_CORRUPTED_METADATA);
+                });
+            });
+
+            // Test from client side viewpoint.
+            final RequestLog log = clientCaptor.get().log().whenComplete().join();
+            assertThat(log.responseStatus()).isEqualTo(HttpStatus.OK);
+            assertThat(log.responseTrailers().get(GrpcHeaderNames.GRPC_STATUS)).isNull();
+            assertThat(log.responseHeaders().get(GrpcHeaderNames.GRPC_STATUS)).satisfies(grpcStatus -> {
+                assertThat(grpcStatus).isNotNull();
+                assertThat(grpcStatus).isEqualTo(String.valueOf(Status.INTERNAL.getCode().value()));
+            });
+            assertThat(log.responseCause())
+                    .isInstanceOf(StatusRuntimeException.class)
+                    .satisfies(cause -> {
+                        assertThat(Status.fromThrowable(cause).getCode()).isEqualTo(
+                                Status.INTERNAL.getCode());
+                        assertThat(Status.trailersFromThrowable(cause)).satisfies(metadata -> {
+                            assertThat(metadata.keys()).doesNotContain(KEY_OF_CORRUPTED_METADATA);
+                        });
+                    });
+
+            // Test from server side viewpoint.
+            final ServiceRequestContextCaptor serviceCaptor = server.requestContextCaptor();
+            assertThat(serviceCaptor.size()).isEqualTo(1);
+            final ServiceRequestContext serviceCtx = serviceCaptor.take();
+            assertThat(serviceCtx.log().ensureComplete().responseCause())
+                    .isInstanceOf(StatusRuntimeException.class)
```

Comment:
What do you think of setting the out of bounds exception to `RequestLog#responseCause`?